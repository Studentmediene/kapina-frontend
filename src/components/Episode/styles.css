.episode {
  background-color: #222;
  color: #DDD;
  cursor: pointer;
  padding: 1rem;
  width: 100%;
  box-shadow:
    -1px 0 2px 0 rgba(0, 0, 0, 0.12),
    1px 0 2px 0 rgba(0, 0, 0, 0.12),
    0 1px 1px 0 rgba(0, 0, 0, 0.24);
  border-radius: 1px;
  margin-bottom: 2rem;
  display: flex;
  flex-direction: row;
}

.playButton {
  padding-right: 11px;
  margin-right: 11px;
  align-self: center;
  display: flex;
  justify-content: center;
  flex-direction: column;
  height: 1rem;
}

.playButtonInner {
  width: 0;
  height: 0;
  border-left: 30px solid #FFF;
  border-top: 15px solid transparent;
  border-bottom: 15px solid transparent;
}

.meta {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
}

.title {
  margin-top: 0.3rem;
}

.lead {
  margin-top: 0.3rem;
  text-align: left;
  color: gray;
}
