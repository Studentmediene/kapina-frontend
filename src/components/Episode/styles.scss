@import '../../variables.scss';

.episode {
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: center;

  background-color: $color-radiorevolt-darkgrey;
  color: $color-text-light;
  padding: 16px;
  width: 100%;
  margin-bottom: 2em;
  cursor: pointer;

  position: relative;
}

.meta {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
<<<<<<< HEAD
=======
  width: 100%;
>>>>>>> made it more responsive
  padding-left: 16px;
  max-width: 90%;
}

.title {
  position: relative;
}

.publishedAt {
  padding-top: 8px;
  color: $color-text-dark;
}

.cropOverflow {
  overflow:hidden; 
  white-space:nowrap; 
  text-overflow:ellipsis;
  max-width: 100%; 
}
