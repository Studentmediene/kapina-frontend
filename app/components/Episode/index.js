/**
*
* Episode
*
*/

import React from 'react';
import moment from 'moment';

import styles from './styles.css';

function getNormalizedDateString(dateString) {
  const paddedString = i => (i < 10 ? `0${i}` : `${i}`);

  const date = moment(dateString);
  const year = date.year();
  const month = date.month() + 1;
  const day = date.date();

  return `${paddedString(day)}.${paddedString(month)}.${year}`;
}

function Episode(props) {
  if (props.digasBroadcastId === 0) {
    return null;
  }

  const playOnDemand = e => {
    e.preventDefault();
    props.playOnDemand(props.id);
  };

  return (
    <button
      className={styles.episode}
      onClick={playOnDemand}
      onKeyPress={playOnDemand}
    >
      <div className={styles.playButton}>
        <div className={styles.playButtonInner} />
      </div>
      <div className={styles.meta}>
        <div className={styles.title}>
          {props.showName} {getNormalizedDateString(props.createdAt)}
        </div>
        <div className={styles.lead}>
          {props.lead}
        </div>
      </div>
    </button>
  );
}

Episode.propTypes = {
  digasBroadcastId: React.PropTypes.number,
  id: React.PropTypes.number,
  title: React.PropTypes.string,
  showName: React.PropTypes.string,
  createdAt: React.PropTypes.string,
  lead: React.PropTypes.string,
  podcastUrl: React.PropTypes.string,
  playOnDemand: React.PropTypes.func,
};

export default Episode;
