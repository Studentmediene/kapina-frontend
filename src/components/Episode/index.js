import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import styles from './styles.scss';
import { togglePlayPause } from 'components/Player/actions';
import PlayPauseButton from 'components/Player/components/PlayPauseButton';
import { selectEpisodeId, selectPaused } from 'components/Player/selectors';

export class Episode extends React.Component {
  static propTypes = {
    digasBroadcastId: PropTypes.number,
    id: PropTypes.number,
    title: PropTypes.string,
    lead: PropTypes.string,
    playOnDemand: PropTypes.func,
  }

  render() {
    if (this.props.digasBroadcastId === 0) {
      return null;
    }

    let isCurrentlyPlaying = false;
  
    if (this.props.playingEpisodeId === this.props.id && !this.props.paused) {
      isCurrentlyPlaying = true;
    }

    const playOnDemand = event => {
      event.preventDefault();
      if(this.props.playingEpisodeId === this.props.id) {
        this.props.togglePlayPause()
      } else {
        this.props.playOnDemand(this.props.id);
      }
    };

    return (
      <div
        className={styles.episode}
        onClick={playOnDemand}
        onKeyPress={playOnDemand}
      >
        <PlayPauseButton
          paused={!isCurrentlyPlaying}
        />
        <div className={styles.meta}>
          <div className={styles.title}>{this.props.title}</div>
          <div className={styles.lead} dangerouslySetInnerHTML={{ __html: this.props.lead }} />
        </div>
      </div>
    );
  }
};

const mapStateToProps = createStructuredSelector({
  playingEpisodeId: selectEpisodeId(),
  paused: selectPaused(),
});

function mapDispatchToProps(dispatch) {
  return {
    togglePlayPause: () => dispatch(togglePlayPause()),
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(Episode);

