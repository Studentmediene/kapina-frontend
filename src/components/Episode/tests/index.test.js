import React from 'react';
import { shallow, mount } from 'enzyme';

import Episode from '../';

describe('<Episode />', () => {
  it('renders correctly', () => {
    const mockProps = {
      digasBroadcastId: 123,
      id: 123,
      title: 'Episode title',
      lead: 'Episode lead',
      playOnDemand: jest.fn(),
    };
    const tree = shallow(<Episode {...mockProps} />);
    expect(tree).toMatchSnapshot();
  });

  it('should return null on invalid digasBroadcastId', () => {
    const mockProps = {
      digasBroadcastId: 0,
    };
    const tree = shallow(<Episode {...mockProps} />);
    expect(tree).toMatchSnapshot();

    const nullEpisode = Episode(mockProps);
    expect(nullEpisode).toBeNull();
  });

  it('calls playOnDemand when clicked', () => {
    const mockProps = {
      digasBroadcastId: 123,
      id: 123,
      title: 'Episode title',
      lead: 'Episode lead',
      playOnDemand: jest.fn(),
    };
    const tree = mount(<Episode {...mockProps} />);
    tree.find('.episode').simulate('click');
    expect(mockProps.playOnDemand).toBeCalled();
  });

  it('calls playOnDemand on keyPress', () => {
    const mockProps = {
      digasBroadcastId: 123,
      id: 123,
      title: 'Episode title',
      lead: 'Episode lead',
      playOnDemand: jest.fn(),
    };
    const tree = mount(<Episode {...mockProps} />);
    tree.find('.episode').simulate('keypress');
    expect(mockProps.playOnDemand).toBeCalled();
  });
});
