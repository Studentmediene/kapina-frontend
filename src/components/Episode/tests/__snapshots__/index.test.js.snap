// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Episode /> renders correctly 1`] = `
<div
  className="episode"
  onClick={[Function]}
  onKeyPress={[Function]}
>
  <PlayPauseButton
    paused={true}
  />
  <div
    className="meta"
  >
    <div
      className="title"
    >
      Episode title
    </div>
    <div
      className="lead"
      dangerouslySetInnerHTML={
        Object {
          "__html": "Episode lead",
        }
      }
    />
  </div>
</div>
`;

exports[`<Episode /> should return null on invalid digasBroadcastId 1`] = `""`;
